akka-camel/src/main/scala/akka/camel/CamelMessage.scala

```diff
@@ -128,6 +129,37 @@ case class CamelMessage(body: Any, headers: Map[String, Any]) {
    */
   def withBodyAs[T](clazz: Class[T])(implicit camelContext: CamelContext): CamelMessage = copy(body = getBodyAs(clazz, camelContext))
 
+  /**
+   * Java API: Creates a new CamelMessageWithAttachments with given <code>attachments</code>. A copy of the attachments map is made.
+   */
+  def withAttachments(attachments: JMap[String, DataHandler]): CamelMessageWithAttachments = new CamelMessageWithAttachments(this.body, this.headers, attachments.toMap)
+
+  /**
+   * Creates a new CamelMessageWithAttachments with given <code>attachments</code>.
+   */
+  def withAttachments(attachments: Map[String, DataHandler]): CamelMessageWithAttachments = new CamelMessageWithAttachments(this.body, this.headers, attachments)
+}
+
+class CamelMessageWithAttachments(override val body: Any, override val headers: Map[String, Any], val attachments: Map[String, DataHandler]) extends CamelMessage(body, headers) {
```

Comment:
instead of `override val` please make these pure constructor parameters that are passed through to the superclass; our convention is to prefix the name with an underscore to ensure that it is not unintentionally used in the class body (which would lead to an additional field being created within the class)